src/components/NavBar/Private/PrivateModal/Private.modal.jsx

```diff
@@ -0,0 +1,23 @@
+import React, { forwardRef } from 'react';
+import ReactDom from 'react-dom';
+
+import PrivateComponent from '../PrivateComponent/Private.component';
+
+import * as Styled from './Private.modal.styled';
+
+const PrivateModal = (props, ref) => {
+  const { isModalOpen } = props;
+
+  if (!isModalOpen) return null;
+
+  return ReactDom.createPortal(
+    <>
```

Comment:
You don't need the Fragments in this case, as you have a root component (`Styled.ModalContainer`)